app/uk/gov/hmrc/agentuserclientdetails/controllers/ClientListController.scala

```diff
@@ -58,6 +58,37 @@ class ClientListController @Inject() (
       }
     }
 
+  def getPaginatedClients(arn: Arn, page: Int = 0, pageSize: Int = 20): Action[AnyContent] = Action.async {
+    implicit request =>
+      withAuthorisedAgent(allowStandardUser = true) { _ =>
+        withGroupIdFor(arn) { groupId =>
+          espConnector
+            .getClientsForGroupId(groupId)
+            .map { clients =>
+              val pageStart = page * pageSize
+              val pageEnd = pageStart + pageSize
+              val currentPageContent = clients.slice(pageStart, Math.min(pageEnd, clients.length - 1))
+              Ok(
+                Json.toJson(
+                  PaginatedList[Client](
+                    pageContent = currentPageContent,
+                    paginationMetaData = PaginationMetaData(
+                      lastPage = false,
```

Comment:
lastPage would be if `pageEnd >= clients.length` ?

Or check currentPageNumber == totalPages might be better
```suggestion
                      lastPage = (clients.length / pageSize) == (page + 1),
```